kernel/src/schema.rs

```diff
@@ -435,8 +440,16 @@ impl DataType {
     pub const TIMESTAMP: Self = DataType::Primitive(PrimitiveType::Timestamp);
     pub const TIMESTAMP_NTZ: Self = DataType::Primitive(PrimitiveType::TimestampNtz);
 
-    pub fn decimal(precision: u8, scale: i8) -> Self {
-        DataType::Primitive(PrimitiveType::Decimal(precision, scale))
+    pub fn decimal(precision: u8, scale: i8) -> DeltaResult<Self> {
+        if precision > 38 || scale > 38 {
```

Comment:
I believe scale cannot legally exceed precision?
```suggestion
        if precision > 38 || scale > precision {
```

Reply:
Aside: why is scale `i8` while precision is `u8`? What would a negative scale even mean?

Reply:
after digging through some more specs / implementations it seems that different databases take wildly different opinions on what valid precisions and scales are. I found this table which summarizes some implementations.

| | Oracle | SQL Server | MySQL | Postgres |
| --- | --- | --- | --- | --- |
Range of precision | 1 to 38 | 1 to 38 | 1 to 65 | 1 to 131,072 |
Range of scale | -84 to 127 | 1 to the value of p | 0 to 30, cannot be larger than p | 0 to 16,383, cannot be larger than p |

from this and the delta spec I would believe following @scovich's suggestion would be the way to go - i.e. `1 <= p <= 38 && s <= p`?